update-format-crau/src/bin/test.rs

```diff
@@ -0,0 +1,89 @@
+use std::io::{Read, Seek, SeekFrom};
+use std::error::Error;
+use std::fs;
+
+use protobuf::Message;
+
+use update_format_crau::proto;
+
+const DELTA_UPDATE_HEADER_SIZE: u64 = 4 + 8 + 8;
+
+#[derive(Debug)]
+struct DeltaUpdateFileHeader {
+    magic: [u8; 4],
+    file_format_version: u64,
+    manifest_size: u64,
+}
+
+impl DeltaUpdateFileHeader {
+    #[inline]
+    fn translate_offset(&self, offset: u64) -> u64 {
+        DELTA_UPDATE_HEADER_SIZE + self.manifest_size + offset
+    }
+}
+
+fn read_delta_update_header(f: &mut dyn Read) -> Result<DeltaUpdateFileHeader, Box<dyn Error>> {
+    let mut header = DeltaUpdateFileHeader {
+        magic: [0; 4],
+        file_format_version: 0,
+        manifest_size: 0,
+    };
+
+    f.read_exact(&mut header.magic)?;
+    if &header.magic != b"CrAU" {
```

Comment:
(Github simply ignored my last comment, so am rewriting)

Ideally let's define `b"CrAU"` as const to be reused elsewhere.

Reply:
sure, no issues. don't think it needs to be a pub export from the update-format-crau subcrate though.

Reply:
Done, defined DELTA_UPDATE_FILE_MAGIC.